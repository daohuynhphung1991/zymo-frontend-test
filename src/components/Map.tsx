import * as React from "react";
import { forwardRef, HTMLAttributes, useMemo, useRef } from "react";
import {
  MapContainer,
  Marker,
  Popup,
  TileLayer,
  MapConsumer,
} from "react-leaflet";
import { positionType } from "types/map";
import { makeStyles } from "@mui/styles";

/**
 * This is a Map component.
 */
const Map = forwardRef<HTMLDivElement, MapProps>(
  ({ label = "", id, zoom = 7, position, makersPosition, ...props }, ref) => {
    const animateRef = useRef(false);

    const classes = useStyles(props);

    return useMemo(
      () => (
        <MapContainer
          center={position}
          zoom={zoom}
          scrollWheelZoom={false}
          className={classes.mapWrapper}
        >
          <TileLayer
            attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />

          <MapConsumer>
            {(map) => {
              if (position)
                map.setView(position, map.getZoom(), {
                  animate: animateRef.current || false,
                });
              return null;
            }}
          </MapConsumer>

          {makersPosition ? (
            makersPosition.length === 1 ? (
              <Marker position={makersPosition[0]}>
                <Popup>
                  A pretty CSS3 popup. <br /> Easily customizable.
                </Popup>
              </Marker>
            ) : (
              makersPosition.map((maker) => {
                return (
                  <Marker position={maker}>
                    <Popup>
                      A pretty CSS3 popup. <br /> Easily customizable.
                    </Popup>
                  </Marker>
                );
              })
            )
          ) : (
            <></>
          )}
        </MapContainer>
      ),
      [position, makersPosition]
    );
  }
);

const useStyles = makeStyles({
  mapWrapper: {
    marginTop: 50,
  },
});

export interface MapProps extends HTMLAttributes<HTMLDivElement> {
  label?: string;
  id?: string;
  zoom?: number;
  position?: positionType;
  makersPosition?: positionType[];
}

export default Map;
