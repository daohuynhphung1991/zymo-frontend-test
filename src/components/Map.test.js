import React from "react";
import renderer from "react-test-renderer";
import Map from "./Map";

test("Map Component", () => {
  const component = renderer.create(<Map>This is Card Component</Map>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
