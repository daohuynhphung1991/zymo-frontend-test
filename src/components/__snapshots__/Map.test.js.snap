// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component 1`] = `
<div
  className="makeStyles-mapWrapper-1"
/>
`;
